integrationtest/java/modules/pojo-standalone-elasticsearch/src/main/java/org/hibernate/search/integrationtest/java/modules/pojo/standalone/elasticsearch/service/AuthorService.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.java.modules.pojo.standalone.elasticsearch.service;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import org.hibernate.search.engine.backend.work.execution.DocumentRefreshStrategy;
+import org.hibernate.search.engine.search.query.SearchQuery;
+import org.hibernate.search.integrationtest.java.modules.pojo.standalone.elasticsearch.entity.Author;
+import org.hibernate.search.mapper.pojo.standalone.loading.SelectionLoadingStrategy;
+import org.hibernate.search.mapper.pojo.standalone.mapping.CloseableSearchMapping;
+import org.hibernate.search.mapper.pojo.standalone.mapping.SearchMapping;
+import org.hibernate.search.mapper.pojo.standalone.session.SearchSession;
+
+public class AuthorService implements AutoCloseable {
+
+	private final Map<Integer, Author> datastore = new HashMap<>();
+	private final CloseableSearchMapping mapping;
+
+	public AuthorService() {
+		mapping = createSearchMapping();
+	}
+
+	private CloseableSearchMapping createSearchMapping() {
+		return SearchMapping.builder()
+				.property( "hibernate.search.automatic_indexing.synchronization.strategy", "sync" )
+				.property( "hibernate.search.backend.log.json_pretty_printing", true )
+				.property( "hibernate.search.backend.analysis.configurer",
+						"org.hibernate.search.integrationtest.java.modules.pojo.standalone.elasticsearch.config.MyElasticsearchAnalysisConfigurer" )
+				.addEntityType(
+						Author.class,
+						context -> context.selectionLoadingStrategy(
+								SelectionLoadingStrategy.fromMap( datastore )
+						)
+				)
+				.build();
+	}
+
+	public void add(String name) {
+		try ( SearchSession session = mapping.createSessionWithOptions()
+				.refreshStrategy( DocumentRefreshStrategy.FORCE )
+				.build() ) {
+
+			Author author = new Author( new Random().nextInt(), name );
```

Comment:
Better hardcode the ID (e.g. `1`); I wouldn't be surprised if some databases failed for negative IDs, or very large IDs.